src/d3a/models/strategy/infinite_bus.py

```diff
@@ -57,17 +60,27 @@ def _set_market_maker_rate(self):
                 ConstSettings.GeneralSettings.DEFAULT_MARKET_MAKER_RATE
 
     def event_activate(self, **kwargs):
-        if self.energy_rate_profile is not None:
+        if self.energy_rate_profile is not None and isinstance(self.energy_rate_profile, dict):
             self.energy_rate = read_and_convert_identity_profile_to_float(self.energy_rate_profile)
             del self.energy_rate_profile
+        elif self.energy_rate_profile is not None and isinstance(self.energy_rate_profile, str):
+            self.energy_rate = read_arbitrary_profile(InputProfileTypes.IDENTITY,
+                                                      self.energy_rate_profile)
+            del self.energy_rate_profile
```

Comment:
Why do we need to delete this attribute?

Reply:
The same question applies to all other deletions.

Reply:
As buying rate irrespective of single value or profile is casted to `self.energy_buy_rate` at the time of `event_activate` and till than it(buying_rate_profile) is saved as instance variable and same goes for selling rate

Reply:
Sorry, I couldn't understand the explanation. `self.buying_rate_profile` contains a string/dict/path that defines a profile file. We process its content and we store the results in `self.energy_buy_rate`.  Why do we need to delete the original `self.buying_rate_profile`?

Reply:
> I mean, I understand that its content might not be used in the remaining operations, but using del on it seems a bit strange (since we never do it in any other situation). Also, this would cause an exception if I try to re-activate the device I suppose.

Can you please shed some clarity where/how you meant to re-activate the device?

Reply:
I'm not planning on doing it, that was just a note on the side. (However, it shows that deleting an instance variable might lead to unexpected effects that are not being handled with proper exceptions).

Reply:
> I'm not planning on doing it, that was just a note on the side. (However, it shows that deleting an instance variable might lead to unexpected effects that are not being handled with proper exceptions).

My intuition was to delete un-needed instance variable(specially this one which contains profile leading to higher memory usage at run-time), but lets be safe than sorry of not thinking about any possible corner case that may fail/crash the simulation

Reply:
I think your intuition is good: if the content of this variable can noticeably increase the memory consumption - and if this can be problematic - then it makes sense to provide ways to avoid it. However, if this is not the case, I think that using `del` is not necessary most of the time. So please proceed as you feel would make more sense 👍